lib/database/user.php

```diff
@@ -378,6 +388,34 @@ function getAccountDetails(&$user, &$dataOut)
     }
 }
 
+function getAccountDetailsFromCookie(?string $cookie, ?array &$dataOut): bool
+{
+    if (empty($cookie)) {
+        return false;
+    }
+
+    sanitize_sql_inputs($cookie);
+
+    // ID, User, and Permissions are used to identify the user and their permissions
+    // DeleteRequested is used to show the "Your account is marked to be deleted" banner
+    // RAPoints, TrueRAPoints, and UnreadMessageCount are used for the logged-in user area
+    // websitePrefs allows pages to enable/disable functionality
+    $query = "SELECT ID, User, Permissions, DeleteRequested,
+                     RAPoints, TrueRAPoints, UnreadMessageCount,
+                     websitePrefs
+                FROM UserAccounts
+                WHERE cookie='$cookie'
+                AND Deleted IS NULL";
+
+    $dbResult = s_mysql_query($query);
+    if ($dbResult == false || mysqli_num_rows($dbResult) !== 1) {
+        return false;
+    }
+
+    $dataOut = mysqli_fetch_array($dbResult);
+    return true;
+}
```

Comment:
To reduce the pass-by-reference usage (it's a overdone in this codebase imo):

```suggestion
function getAccountDetailsFromCookie(?string $cookie): ?array
{
    if (empty($cookie)) {
        return null;
    }

    sanitize_sql_inputs($cookie);

    // ID, User, and Permissions are used to identify the user and their permissions
    // DeleteRequested is used to show the "Your account is marked to be deleted" banner
    // RAPoints, TrueRAPoints, and UnreadMessageCount are used for the logged-in user area
    // websitePrefs allows pages to enable/disable functionality
    $query = "SELECT ID, User, Permissions, DeleteRequested,
                     RAPoints, TrueRAPoints, UnreadMessageCount,
                     websitePrefs
                FROM UserAccounts
                WHERE cookie='$cookie'
                AND Deleted IS NULL";

    $dbResult = s_mysql_query($query);
    if (!$dbResult || mysqli_num_rows($dbResult) !== 1) {
        return null;
    }

    return mysqli_fetch_array($dbResult) ?: null;
}
```